client/app/components/Link.tsx

```diff
@@ -14,6 +15,18 @@ function Link({ children, ...props }: LinkProps) {
   return <Link.Component {...props}>{children}</Link.Component>;
 }
 
+type LinkButtonProps = PlainButtonProps | LinkProps;
+
+function LinkOrButton(props: LinkButtonProps) {
```

Comment:
This feels too magical to me. I feel like our markup should stay stable -- if a developer wants to display a link, they should display a link. If they want to display a button, they can display a button.

Reply:
This component is specifically intended for cases in which the developer doesn't know what he needs to display in advance. I'll push changes that better explain that.

Reply:
Maybe we can improve its usage by requiring that both href and onClick attributes be set? WDYT?

Reply:
I think the developer should specify inline which one needs to be displayed. Ideally, our markup should be stable. 

For example, if we're trying to write a test for a page that has some interactive element, we'll now need to look at the code to determine whether or not we should query for a `link` role or a `button` role, right?

Can you show me some concrete places where we're having this issue?

Reply:
Just added them. I don't think there are any others as of now. The problem stated was present before either way, as an `<a>` rendered without an `href`, which was the case, would result not only in an inaccessible link, but also in one that lacks link role.

Reply:
Considering:
1. there are very few cases where it can be both a link or a button depending on the parameters
2. we want to encourage the discerning of a Button and a Link (that by the fact that you initially separated those in different components)

I think creating a new component is not what we want here. I'd rather leave this responsibility to the behavior of each component that might require this "exceptional" behavior.